apps/server/src/modules/tool/external-tool/uc/external-tool.uc.ts

```diff
@@ -16,17 +23,34 @@ export class ExternalToolUc {
 
 	async createExternalTool(userId: EntityId, externalToolCreate: ExternalToolCreate): Promise<ExternalTool> {
 		const externalTool = new ExternalTool({ ...externalToolCreate });
+		await this.addLogoToExternalToolIfExists(externalTool);
 
 		await this.ensurePermission(userId, Permission.TOOL_ADMIN);
 		await this.toolValidationService.validateCreate(externalTool);
 
-		const tool: Promise<ExternalTool> = this.externalToolService.createExternalTool(externalTool);
+		const tool: ExternalTool = await this.externalToolService.createExternalTool(externalTool);
 
 		return tool;
 	}
 
+	private async addLogoToExternalToolIfExists(externalTool: Partial<ExternalTool>): Promise<Partial<ExternalTool>> {
+		if (!externalTool.logoUrl) {
+			return externalTool;
+		}
+
+		const base64Logo: string | null = await this.externalToolService.fetchBase64Logo(externalTool.logoUrl);
+		if (base64Logo) {
+			externalTool.logoBase64 = base64Logo;
```

Comment:
Please clean up this wording in entity and service. it should be either base64Logo or logoBase64 to make sure that the same thing is meant.
(I would prefer logoBase64, or drop Base64 completely, as this is an implementation detail IMO)